### DISCUSSION TITLE
Loops and crashes - Never ever do this!

### DISCUSSION THREAD

**Discussion by jenstornell** [tailwindlabs/tailwindcss#8023] [General]
**Dangerous setup**

Never ever do like below. I did and does not work like you might expect.

```
module.exports = {
  content: ["./**/*.js"],
};
```

So, what does the above do? It searched through all javascript files for Tailwind classes. So far so good. However, there is a file called `tailwind.config.js`. Note that it's also a javascript file and will be included.

**So what happens when running this?**

- Before it took like 200-500 ms to run. Now it takes 16 seconds.
- It says that the CSS is valid but it's not. Some parts of the CSS file stops working.
- I also ran out of memory a few times.

**Comment by adamwathan** [comment]
The main problem here is that that pattern includes `node_modules`, so you are are scanning every file from every dependency in your project. This is covered in the documentation as well:

https://tailwindcss.com/docs/content-configuration#pattern-recommendations

<img width="816" alt="image" src="https://user-images.githubusercontent.com/4323180/161386400-5ec32d49-91f3-44b4-8e2f-00a92b9aca4c.png">
